app/src/main/java/com/github/se/polyfit/ui/screen/RecipeRecommendationScreen.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.se.polyfit.ui.screen
+
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.core.content.ContextCompat
+import androidx.navigation.NavHostController
+import com.github.se.polyfit.R
+import com.github.se.polyfit.model.recipe.Recipe
+import com.github.se.polyfit.ui.components.GenericScreen
+import com.github.se.polyfit.ui.components.recipe.RecipeCard
+
+@Composable
+fun RecipeRecommendationScreen(navController: NavHostController, recipes: List<Recipe>) {
```

Comment:
You shouldn't pass directly the navController, but a function of it. [More explication](https://developer.android.com/guide/navigation/use-graph/navigate)

Reply:
The generic screens need a `navController`. Yeah, directly passing it is not ideal. We should fix this in a later PR, if we have time. I'll need to pass navigation functions in the implementation that creates the view model, and I will follow your advice.